core/comment_events.js

```diff
@@ -399,15 +495,19 @@ Blockly.Events.CommentMove.prototype.isNull = function() {
  * @param {boolean} forward True if run forward, false if run backward (undo).
  */
 Blockly.Events.CommentMove.prototype.run = function(forward) {
-  var workspace = this.getEventWorkspace_();
-  var comment = workspace.getCommentById(this.commentId);
+  var comment = this.getComment_();
   if (!comment) {
     console.warn('Can\'t move non-existent comment: ' + this.commentId);
     return;
   }
 
   var target = forward ? this.newCoordinate_ : this.oldCoordinate_;
-  // TODO: Check if the comment is being dragged, and give up if so.
-  var current = comment.getXY();
-  comment.moveBy(target.x - current.x, target.y - current.y);
+
+  if (comment instanceof Blockly.ScratchBlockComment) {
+    comment.moveTo(target.x, target.y);
+  } else {
+    // TODO: Check if the comment is being dragged, and give up if so.
```

Comment:
Does this TODO apply to both, or only workspace comments?

Do workspace comments need a `moveTo` function as well as a `moveBy` function?

Reply:
I wasn't sure about this TODO, it was left over from before I added the instanceof check. I'm not sure what scenario would lend itself to this code getting run while the comment is being dragged...

Reply:
Okay, we can leave it and I can come back and try to figure out if that's discussing a valid problem.  If not I'll delete it.